base_custom_attributes/custom_attributes.py

```diff
@@ -0,0 +1,449 @@
+            etree.SubElement(
+                parent,
+                'separator',
+                string="%s" % attribute.field_description,
+                colspan="4")
+            kwargs['nolabel'] = "1"
+        if attribute.ttype in ['many2one', 'many2many']:
+            if attribute.relation_model_id:
+                # attribute.domain is a string, it may be an empty list
+                try:
+                    domain = ast.literal_eval(attribute.domain)
+                except ValueError:
+                    domain = None
+                if domain:
+                    kwargs['domain'] = attribute.domain
+                else:
+                    ids = [op.value_ref.id for op in attribute.option_ids]
+                    kwargs['domain'] = "[('id', 'in', %s)]" % ids
+            else:
+                kwargs['domain'] =\
+                    "[('attribute_id', '=', %s)]" % attribute.attribute_id.id
+        kwargs['context'] =\
+            "{'default_attribute_id': %s}" % attribute.attribute_id.id
+        kwargs['required'] = str(attribute.required or
+                                 attribute.required_on_views)
+        field = etree.SubElement(parent, 'field', **kwargs)
+        orm.setup_modifiers(field, self.fields_get(attribute.name))
+        return parent
+
+    @api.model
+    def _build_attributes_notebook(self, attribute_group_ids):
+        notebook = etree.Element('notebook', name="attributes_notebook",
+                                 colspan="4")
+        toupdate_fields = []
+        grp_obj = self.env['attribute.group']
+        for group in grp_obj.browse(attribute_group_ids):
+            page = etree.SubElement(notebook, 'page',
+                                    string=group.name.capitalize())
+            for attribute in group.attribute_ids:
+                if attribute.name not in toupdate_fields:
+                    toupdate_fields.append(attribute.name)
+                    self._build_attribute_field(page, attribute)
+        return notebook, toupdate_fields
+
+    @api.onchange('relation_model_id')
+    def relation_model_id_change(self):
+        "removed selected options as they would be inconsistent"
+        return {'value': {'option_ids': [(2, i[1]) for i in self.option_ids]}}
+
+    @api.multi
+    def button_add_options(self):
+        self.ensure_one()
+        wizard_form = self.env.ref(
+            'base_custom_attributes.attribute_option_wizard_form_view', False)
+        return {
+            'name': _('Options Wizard'),
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'attribute.option.wizard',
+            'type': 'ir.actions.act_window',
+            'views': [(wizard_form.id, 'form')],
+            'view_id': wizard_form.id,
+            'target': 'new',
+        }
+
+    @api.model
+    def _get_default_model(self):
+        context = self._context
+        if context and context.get('force_model'):
```

Comment:
It is enough to write

    if context.get('force_model'):

as context is always a frozen dict even if empty